admin/app/components/solidus_admin/ui/table/component.js

```diff
@@ -42,10 +42,13 @@ export default class extends Controller {
     }
   }
 
-  showSearch(event) {
+  showSearch({ detail: { focus } }) {
     this.modeValue = "search"
     this.render()
-    this.searchFieldTarget.focus()
+
+    if (focus ?? true) {
```

Comment:
I'm having a difficult time understanding this at first sight, I think the same can be achieved with `focus == null` or `focus !== false`. A deeper change would be changing the detail property, turning `focus` into `avoidFocus` and letting the default value clearly distinguished from the explicit value.